### PR TITLE
Cleanup syncset secretreference secrets when ResourceApplyMode==Sync

### PR DESCRIPTION
**Pull request by abutcher**
https://jira.coreos.com/browse/CO-587

### CODE REVIEW

File: pkg/controller/syncsetinstance/syncsetinstance_controller.go [hunk 1]
```diff
@@ -360,10 +360,14 @@ func (r *ReconcileSyncSetInstance) syncDeletedSyncSetInstance(ssi *hivev1.SyncSe
...
 	ssiLog.Info("deleting syncset resources on target cluster")
 	err = r.deleteSyncSetResources(ssi, dynamicClient, ssiLog)
-	if err == nil {
-		return reconcile.Result{}, r.removeSyncSetInstanceFinalizer(ssi, ssiLog)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	err = r.deleteSyncSetSecretReferences(ssi, dynamicClient, ssiLog)
+	if err != nil {
+		return reconcile.Result{}, err
```
Comment (LalatenduMohanty) [inline pkg/controller/syncsetinstance/syncsetinstance_controller.go]: @abutcher Is it a safe assumption that secrets are the only objects that need to get attached to syncsets?

Comment (abutcher) [reply pkg/controller/syncsetinstance/syncsetinstance_controller.go]: Generic resources (objects) and secrets thus far.

File: pkg/controller/syncsetinstance/syncsetinstance_controller.go [hunk 2]
```diff
@@ -428,6 +432,36 @@ func (r *ReconcileSyncSetInstance) deleteSyncSetResources(ssi *hivev1.SyncSetIns
...
+		if err != nil {
+			switch {
+			case errors.IsNotFound(err):
+				secretLog.Debug("secret not found, nothing to do")
+			case errors.IsForbidden(err):
+				secretLog.WithError(err).Error("forbidden secret deletion, skipping")
+			default:
+				lastError = err
+				secretLog.WithError(err).Error("error deleting secret")
+				ssi.Status.SecretReferences[index].Conditions = r.setDeletionFailedSyncCondition(ssi.Status.SecretReferences[index].Conditions, fmt.Errorf("failed to delete secret: %v", err))
```
Comment (staebler) [inline pkg/controller/syncsetinstance/syncsetinstance_controller.go]: The controller never updates the status, so these conditions are lost.

Comment (staebler) [reply pkg/controller/syncsetinstance/syncsetinstance_controller.go]: It looks like the same things is done when deleting the resources.

**Pull request merged**